openml/extensions/sklearn/extension.py

```diff
@@ -1268,154 +1249,50 @@ def _prediction_to_probabilities(
             user_defined_measures['wall_clock_time_millis'] = (modelfit_dur_walltime
                                                                + modelpredict_duration_walltime)
 
-        # add client-side calculated metrics. These is used on the server as
-        # consistency check, only useful for supervised tasks
-        def _calculate_local_measure(sklearn_fn, openml_name):
-            user_defined_measures[openml_name] = sklearn_fn(test_y, pred_y)
-
-        # Task type specific outputs
-        arff_datacontent = []
-
         if isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
 
+            if classes is None:
```

Comment:
I would personally prefer argument checking to be done at the start of the method call such that all constraints are clear, and `TypeError` lead to a quick fail (not e.g. after running a whole grid search!). Even though I know it introduces an extra line of code :)

Reply:
That check will go away when using `task.class_labels`. I will add a check that `X_test` and `y_train` are given at the top of the function.